src/Tasks/Copy.cs

```diff
@@ -361,7 +361,7 @@ private void LogDiagnostic(string message, params object[] messageArgs)
         private void TryCopyViaLink(string linkComment, MessageImportance messageImportance, FileState sourceFileState, FileState destinationFileState, out bool linkCreated, ref string errorMessage, Func<string, string, string, bool> createLink)
         {
             // Do not log a fake command line as well, as it's superfluous, and also potentially expensive
-            Log.LogMessage(MessageImportance.Normal, linkComment, sourceFileState.Name, destinationFileState.Name);
+            Log.LogMessage(MessageImportance.Normal, linkComment, sourceFileState.FileNameFullPath, destinationFileState.FileNameFullPath);
```

Comment:
Is `FileNameFullPath` allways guaranteed to be populated here?
Should we coallesce with `Path.GetFullPath(<fileState>.Name)` or at least `<fileState>.Name`?

Reply:
I aligned this with the existing code: https://github.com/dotnet/msbuild/blob/0125fc9fb838951c8750135c71336aa1f5e868a9/src/Tasks/Copy.cs#L341

But to answer the question, yes it's always guaranteed. The full path is populated in this call:

https://github.com/dotnet/msbuild/blob/9ae833e5fc229f521fc017d0ece4b9353289d699/src/Tasks/Copy.cs#L756

The actual copy that will try to utilize the full path happens here: https://github.com/dotnet/msbuild/blob/9ae833e5fc229f521fc017d0ece4b9353289d699/src/Tasks/Copy.cs#L767

Reply:
Perfect - thanks for confirming